content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,362 @@
+  provider.addSpanProcessor(new SimpleSpanProcessor(consoleExporter))
+  // send spans to Jaeger or any OpenTelemetry compliant collector
+  provider.addSpanProcessor(new SimpleSpanProcessor(otlpTraceExporter))
+
+  provider.register()
+
+  // Enable tracing in Prisma
+  registerInstrumentations({
+    instrumentations: [new PrismaInstrumentation()],
+  })
+}
+
+//set up tracing for this application
+otelSetup()
+
+// set things up for tracing our "app"
+const tracer = trace.getTracer('Example Application')
+
+async function main() {
+  const prisma = new PrismaClient()
+
+  let users = await tracer.startActiveSpan('create-user', async (span) => {
+    try {
+      const email = `user.${Date.now()}@sample-application.com`
+      await prisma.user.create({
+        data: {
+          email: email,
+        },
+      })
+      let users = await prisma.user.findMany()
+      return users
+    } finally {
+      span.end()
+    }
+  })
+
+  console.table(users)
+}
+
+main().then(() => console.log('example complete'))
+```
+
+## Trace middleware
+
+If you want to include your [middleware](/concepts/components/prisma-client/middleware) in your tracing, then set `middleware` to `true` in your `registerInstrumentations` statement:
+
+```ts
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation({ middleware: true })],
+})
+```
+
+## Send tracing to the console
+
+To send tracing to the console, add the following to you application code:
+
+```ts
+import { context } from '@opentelemetry/api'
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { Resource } from '@opentelemetry/resources'
+import {
+  BasicTracerProvider,
+  SimpleSpanProcessor,
+  ConsoleSpanExporter,
+} from '@opentelemetry/sdk-trace-base'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { PrismaInstrumentation } from '@prisma/instrumentation'
+import { PrismaClient } from '@prisma/client'
+
+const contextManager = new AsyncHooksContextManager().enable()
+
+context.setGlobalContextManager(contextManager)
+
+const provider = new BasicTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'test-tracing-service',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+
+// Console exporter
+const consoleExporter = new ConsoleSpanExporter()
+provider.addSpanProcessor(new SimpleSpanProcessor(consoleExporter))
+provider.register()
+
+// Register Prisma Instrumentation
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation()],
+})
+
+async function main() {
+  const prisma = new PrismaClient()
+
+  const email = `user.${Date.now()}@prisma.io`
+
+  await prisma.user.create({
+    data: {
+      email: email,
+    },
+  })
+}
+
+main()
+```
+
+## Send tracing to Jaeger
+
+To send tracing to Jaeger, add the following to you application code:
```

Comment:
Maybe add quick introduction what Jaeger is or why it is useful (can be run locally, free)

Reply:
We've already introduced Jaeger, along with some other tracing systems, and linked to its official site, up-page. I'll expand the intro.